spark2.4/src/main/scala/KafkaReader.scala

```diff
@@ -45,23 +47,33 @@ object KafkaReader {
 //
 //    val metricsProducer = new KafkaProducer[Integer, Long](props)
 //    val metricsTopic = "ssds-metrics"
-
+    
+    // Create schema from ddl string like
+    //    "ID STRING NOT NULL, LOCATION STRING, TEMPERATURE DOUBLE, HUMIDITY DOUBLE, TM TIMESTAMP"
+    var schema = new StructType
+    schemaDDL.split(",").foreach( s => {
+      val f = s.trim.split(" ")
+      schema = schema.add( f(0) , f(1) , ! s.toUpperCase.contains("NOT NULL") )
+    })
+
+    // Keep for reference -- Weather demo
     //    val schema = StructType(
 //      StructField("ID", StringType, false) ::
 //      StructField("LOCATION", StringType, true) ::
 //      StructField("TEMPERATURE", DoubleType, true) ::
 //      StructField("HUMIDITY", DoubleType, true) ::
 //      StructField("TM", TimestampType, true) :: Nil)
 
-    val schema = StructType(
-      StructField("ID", LongType, false) ::
-      StructField("PAYLOAD", StringType, true) ::
-      StructField("SRC_SERVER", StringType, false) ::
-      StructField("SRC_THREAD", LongType, true) ::
-      StructField("TM_GENERATED", LongType, false) :: Nil)
-//      StructField("PTN_NSDS", IntegerType, true) ::
-//      StructField("TM_NSDS", LongType, true) :: Nil)
-//      StructField("TM_EXT_KAFKA", LongType, true) :: Nil)
+    // Keep for reference -- Performance test (first five fields)
+//    val schema = StructType(
+//      StructField("ID", LongType, false) ::
+//      StructField("PAYLOAD", StringType, true) ::
+//      StructField("SRC_SERVER", StringType, false) ::
+//      StructField("SRC_THREAD", LongType, true) ::
+//      StructField("TM_GENERATED", LongType, false) :: Nil)
+////      StructField("PTN_NSDS", IntegerType, true) ::
+////      StructField("TM_NSDS", LongType, true) :: Nil)
+////      StructField("TM_EXT_KAFKA", LongType, true) :: Nil)
```

Comment:
Please remove all the lines (they're in git anyway).

Reply:
Removed in commit https://github.com/splicemachine/splice-machine-spark-connector/commit/43f72a64c5a3c29c4609801af6e8f428e4fde6aa